File: output/ChatGPT/avatar/Java/Python/atcoder_ABC152_C.py
import heapq

def solve():
    N = int(input())
    arr = list(map(int, input().split()))
    sum = 0
    q = []
    for i in range(len(arr)):
        heapq.heappush(q, arr[i])
        if arr[i] <= q[0]:
            sum += 1
    return sum

print(solve())

File: output/ChatGPT/avatar/Java/C++/atcoder_ABC152_C.cpp
#include <iostream>
#include <queue>
using namespace std;

int solve() {
    int N;
    cin >> N;
    int arr[N];
    for (int i = 0; i < N; i++) {
        cin >> arr[i];
    }
    int sum = 0;
    priority_queue<int> q;
    for (int i = 0; i < N; i++) {
        q.push(arr[i]);
        if (arr[i] <= q.top()) {
            sum++;
        }
    }
    return sum;
}

int main() {
    cout << solve() << endl;
}